google/cloud/bigquery/table.py

```diff
@@ -452,17 +453,20 @@ def schema(self):
                 instance or a compatible mapping representation of the field.
         """
         prop = self._properties.get(self._PROPERTY_TO_API_FIELD["schema"])
-        if not prop:
+        if not prop:  # if empty Schema, empty list, None
             return []
-        else:
-            return _parse_schema_resource(prop)
+        elif isinstance(prop, Schema):
```

Comment:
This would be concerning if this were ever true. `_properties` should always be equivalent to the `REST` representation.

Reply:
@tswast Question for you.

This response relates to setting the `schema` on the table as a whole and thus applies to several comments in this review:

In general, I see the reasoning behind maintaining the `_properties` attribute as a dict that is pretty close to the API resource representation.

If the user provides a `Schema` object based on the `Schema` class, does it make sense to maintain this as a `Schema` object until we need to explicitly convert it back into an API resource using `.to_api_repr()`?

Reply:
If so, I wouldn't put it in `_properties`.

Ideally, the objects we provide would just be a "view" on top of the REST resource (`_properties`). I wouldn't put anything that requires conversion before sending it to the BigQuery API into such objects.

In some cases, we have to store state outside of `_properties`, in which case we populate the necessary dictionaries and such in `to_api_repr()`. The `UserList` approach may be such a case. If so, then don't store the backing list of `SchemaField` in `_properties`. Store it in some other instance variable.